S3/Config.py

```diff
@@ -275,38 +281,46 @@ def role_refresh(self):
             except:
                 warning("Could not refresh role")
 
-    def env_config(self):
-        cred_content = ""
+    def aws_credential_file(self):
         try:
-            cred_file = open(os.environ['AWS_CREDENTIAL_FILE'],'r')
-            cred_content = cred_file.read()
+            aws_credential_file = os.path.expanduser('~/.aws/credentials') 
+            if 'AWS_CREDENTIAL_FILE' in os.environ and os.path.isfile(os.environ['AWS_CREDENTIAL_FILE']):
+                aws_credential_file = os.environ['AWS_CREDENTIAL_FILE']
```

Comment:
I know that it was not done before, but as an improvement, it would be great if you can ensure to pass the strings you get from os.environ to config_unicodise.
The tests to be done to check if things are good, is like to create a folder with chinese characters, and try to put some chinese chars inside the configuration file itself. Normally keys don't have special chars, but so we are sure that the code is proof to nasty issues in the future.